### DISCUSSION TITLE
nlp.create_pipe() Config Validation Error

### DISCUSSION THREAD

**Discussion by WuJiunShiung** [explosion/spaCy#7068] [Help: Coding & Implementations] [answered]
Hi, Guys,

I am using spaCy 3.0 on Win 10 (64 bit). I am watching TweetTexClassificationWithSpacy on Youtube.

I encountered Config Validation Error when I tried the following:
nlp = spacy.load("en_core_web_md")
textcat = nlp.create_pipe("textcat", config={"exlcusive_classes":True, "architecture":"bow"})

![image](https://user-images.githubusercontent.com/11987452/107902701-2610d100-6f82-11eb-8158-a722f2865177.png)

Could someone help me with this and/or point me to any relevant documentation? Thanks!

**Comment by svlandeg** [comment, marked as answer]
The tutorial/video you're following seems to be for spaCy 2.x? We've [changed](https://spacy.io/usage/v3) a few things in the configuration mechanism for 3.0, so this code will require a little bit of adjusting.

You can find the dictionary keys for the `textcat` config here: https://spacy.io/api/textcategorizer#config. There are only two fields, `model` and `threshold`:
```
from spacy.pipeline.textcat import DEFAULT_SINGLE_TEXTCAT_MODEL
config = {
   "threshold": 0.5,
   "model": DEFAULT_SINGLE_TEXTCAT_MODEL,
}
nlp.add_pipe("textcat", config=config)
```
If you want the BOW model specifically, you can do something like this:
```
    model = {
        "@architectures": "spacy.TextCatBOW.v1",
        "exclusive_classes": True,
        "ngram_size": 2,
        "no_output_layer": False,
    }
    config = {"model": model}
    textcat = nlp.add_pipe("textcat", config=config)
```
Hope that works for you!

**Reply by WuJiunShiung** [reply]
Thank you very much!
